### PR TITLE
Sms customization docs

### PR DESCRIPTION
**Pull request by MaxwellLevenson**

### CODE REVIEW

File: ingredients/sms_customization/SMS_Customization_Doc.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+## SMS Customization
```
Comment (dmitrig01) [inline ingredients/sms_customization/SMS_Customization_Doc.md]: let's keep the filename consistent with everything else - no _Doc, all lower case :-)

File: recipes/text_me_the_app_page.md [hunk 2]
```diff
@@ -28,3 +28,6 @@ The Branch Web SDK gives you the ability to quickly build a text-me-the-app feat
 
 {% ingredient web_sdk/send_sms %}{% endingredient %}
 {% ingredient web_sdk/send_sms_example %}{% endingredient %}
+
+## Customizing SMS Messages
```
Comment (dmitrig01) [inline recipes/text_me_the_app_page.md]: you have the title once in the page, and once here -- this'll show twice. did you test this :-)?

File: ingredients/sms_customization/SMS_Customization_Doc.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+To customize SMS install messages, go to the *Text me the app page* section of the settings tab under general.
+![SMS Customization Tags](http://i.imgur.com/giQCIJu.png)
```
Comment (derrickstaten) [inline ingredients/sms_customization/SMS_Customization_Doc.md]: please reference pictures in repo. also, take a look at the {% image %} tags used in many places

File: ingredients/sms_customization/SMS_Customization_Doc.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+There are two possibilities for the SMS messages. Either:
+
+1) You can define a special message based on your own parameters in the *Deep Link Data* section at the bottom of the link editing screen.
+![Deep Link Data](http://i.imgur.com/j6WP2yh.png)
```
Comment (derrickstaten) [inline ingredients/sms_customization/SMS_Customization_Doc.md]: ditto

File: ingredients/sms_customization/SMS_Customization_Doc.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+Use the *key* of *custom\_sms\_text* and then enter your custom value in the value section. This will be added before your Branch link in the message.
+
+2) The SMS will default to "Click to download here --" followed by your Branch link. If you wish to customize the default message, edit the message following the keyword *default* in the *Text me the app page*.
+![Default Text](http://i.imgur.com/YVfdHRw.png)
```
Comment (derrickstaten) [inline ingredients/sms_customization/SMS_Customization_Doc.md]: ditto

**Comment by dmitrig01** [comment]
this shows effectively where to change the text, but not what to change the text to -- it should talk about what to put inside the tags, etc

**Comment by dmitrig01** [comment]
also, I don't think we actually support custom_sms_text, do we? we should add it

File: ingredients/sms_customization/SMS_Customization_Doc.md (hunk 1 above)
Comment (derrickstaten) [inline ingredients/sms_customization/SMS_Customization_Doc.md]: also, can we locate this within an existing ingredients grouping? ingredients/dashboard_setup/sms_customization.md would be a good way to go

File: recipes/text_me_the_app_page.md (hunk 2 above)
Comment (derrickstaten) [inline recipes/text_me_the_app_page.md]: please test this before pinging us again -- thanks!

**Comment by MaxwellLevenson** [comment]
Isn't custom_sms_text just another parameter passed into link.data? About what to put in the tags e.g. what?

**Comment by MaxwellLevenson** [comment]
@dmitrig01, @derrickstaten This should look a lot better.

File: ingredients/dashboard_setup/sms_customization.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+####3. Branch Default
+If a developer doesn't specify a link specific message or a custom default, SMS messages will default to the format of  
+`"Click here to download [App Name] {{ link }}"`
+
+
+#### Reference:
+- The tag _\{\{ link \}\}_ is replaced with your Branch link when sent
+- The key `$custom_sms_text` is used for creating link specific custom SMSs
+- *Advanced:* In both the link specific custom messages and the custom default message, there is support for other liquid tags like _\{\{ link.channel \}\}_ or _\{\{ link.campaign \}\}_
```
Comment (dmitrig01) [inline ingredients/dashboard_setup/sms_customization.md]: can we expand this into it's whole own section with samples, and show specifically a link.data.X sample, like

"suppose oyu create links for your blog posts. each one has a title, so you put `{ title: "foo" }` in the link data. you can set your SMS text to 'Click to see {{ link.data.title }}: {{ link }}'"

except more better

@derrickstaten ^^^ this is the reason why I think we need to allow app preview templates to allow any parameter: for consistency here and other places :-)

File: ingredients/dashboard_setup/sms_customization.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
+Branch has several options for the customization of SMS messages from the Text Me the App page.
+
+The following are the three different types of messages that can be sent:
+
+####1. Link Specific Messages
+You can define a special SMS message one the individual link scale. Whether you want to switch the language of a message for a different region or include device specific date, you can specify the message in the *Deep Link Data* section at the bottom of the link editing screen.  
+![Deep Link Data](/img/ingredients/dashboard_setup/deeplink_data.png)
```
Comment (dmitrig01) [inline ingredients/dashboard_setup/sms_customization.md]: let's be sure to show that you still need to put {{ link }} in the screenshot as well :-)

File: ingredients/dashboard_setup/sms_customization.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+For ads in Spain:  
+`Haz click aquí para descargar FlowerPower {{ link }}`
+
+For ads in Germany:  
+`Klicken Sie auf das FlowerPower hier herunterladen {{ link }}`
+
+####2. Custom Default for All Messages
+You can create your own custom default message that will be sent if the specific link someone clicks doesn't have a customized message itself. To customize the default message, edit the form under the *Text me the app page* tab in the general settings	.
+![Default Text](/img/ingredients/dashboard_setup/default_message.png)
```
Comment (dmitrig01) [inline ingredients/dashboard_setup/sms_customization.md]: this one's out of date :-)

File: ingredients/dashboard_setup/sms_customization.md (hunk 6 above)
Comment (dmitrig01) [inline ingredients/dashboard_setup/sms_customization.md]: you'll also want to talk about the | default syntax and where it's useful

Comment (derrickstaten) [inline ingredients/dashboard_setup/sms_customization.md]: @dmitrig01 we are going to allow that. there just won't be any mapping. devs can customize private templates to use whatever key-value pairs they send up.

basically: you're right

**Comment by MaxwellLevenson** [comment]
@dmitrig01

**Comment by dmitrig01** [comment]
looks godo to me - @derrickstaten

**Comment by dmitrig01** [comment]
@MaxwellLevenson can you merge the latest master into your branch, so we don't have mereg conflicts?

**Comment by MaxwellLevenson** [comment]
Just merged in master

**Comment by aaustin** [comment]
:+1:

**Pull request merged**